src/lib/commandBarConfigs/namedViewsConfig.ts

```diff
@@ -0,0 +1,222 @@
+import { NamedView } from 'wasm-lib/kcl/bindings/NamedView'
+import { Command } from '../commandTypes'
+import toast from 'react-hot-toast'
+import { engineCommandManager } from 'lib/singletons'
+import { uuidv4 } from 'lib/utils'
+import { settingsActor } from 'machines/appMachine'
+import { reportRejection } from 'lib/trap'
+
+export function createNamedViewsCommand() {
+  // Creates a command to be registered in the command bar.
+  // The createNamedViewsCommand will prompt the user for a name and then
+  // hit the engine for the camera properties and write them back to disk
+  // in project.toml.
+  const createNamedViewCommand: Command = {
+    name: 'Create named view',
+    displayName: `Create named view`,
+    description:
+      'Saves a named view based on your current view to load again later',
+    groupId: 'namedViews',
+    icon: 'settings',
+    needsReview: false,
+    onSubmit: (data) => {
+      const invokeAndForgetCreateNamedView = async () => {
+        if (!data) {
+          return toast.error('Unable to create named view, missing name')
+        }
+
+        // Retrieve camera view state from the engine
+        const cameraGetViewResponse =
+          await engineCommandManager.sendSceneCommand({
+            type: 'modeling_cmd_req',
+            cmd_id: uuidv4(),
+            // @ts-ignore TODO: Not in production yet.
+            cmd: { type: 'default_camera_get_view' },
+          })
+
+        if (!(cameraGetViewResponse && 'resp' in cameraGetViewResponse)) {
+          return toast.error('Unable to create named view, websocket failure')
+        }
+
+        if ('modeling_response' in cameraGetViewResponse.resp.data) {
+          // @ts-ignore TODO: Not in production yet.
+          const view = cameraGetViewResponse.resp.data.modeling_response.data
+          // Create a new named view
+          const requestedView: NamedView = {
+            name: data.name,
+            ...view.view,
+          }
+          // Retrieve application state for namedViews
+          const namedViews = [
+            ...settingsActor.getSnapshot().context.app.namedViews.current,
+          ]
+
+          // Create and set namedViews application state
+          const requestedNamedViews = [...namedViews, requestedView]
+          settingsActor.send({
+            type: `set.app.namedViews`,
+            data: {
+              level: 'project',
+              value: requestedNamedViews,
+            },
+          })
+          toast.success(
+            `Your named view ${requestedView.name} successfully created.`
+          )
```

Comment:
Is it possible that the send fails? Because then we won't want to show a success dialog.

Edit: Kevin pointed out all our settings have this deficiency